### PR TITLE
Rework the PGP key configuration

### PR DESCRIPTION
**Pull request by kinnison**
This PR adds a number of commits which work toward a better PGP key infrastructure inside Rustup.

In particular it adds (deliberately undocumented at this time) a `settings.toml` entry `pgp_keys` and the environment variable `RUSTUP_PGP_KEY`.

We add a `PgpKey` enum which indicates where a key came from, so that we can use it in future debugging and/or informational messages to the user.

Perhaps @dignifiedquire might take a look over this?

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -33,6 +33,41 @@ impl Display for OverrideReason {
     }
 }
 
+#[derive(Debug)]
+pub enum PgpKey {
```
Comment (dignifiedquire) [inline src/config.rs]: should probably document if the intention is for these to be public or secret keys

**Review by dignifiedquire — APPROVED** [review]
Nice, this should do the trick. 

Is there a good place to document this though? As currently this seems a bit hidden for a user

File: src/config.rs (hunk 1 above)
Comment (kinnison) [reply src/config.rs]: I agree, I'll push a change momentarily to rename that to `PgpPublicKey` to make it clearer.

**Comment by kinnison** [comment]
I am deliberately hiding this for now.  Once we have proper signature checking integrated, I'll probably add something to the `README.md` about public keys, since I'll probably add a `rustup show keys` or somesuch to enumerate and show all the keys rustup knows about and where they come from.

**Pull request merged**